# Análise Inicial — Turnover RH

Discovery das bases e definição do universo analítico.

**Bases principais:**
- `FtDemitidosTurnoverRH` — eventos de desligamento
- `FtFuncionarioRH_amostra` — painel mensal de colaboradores

In [1]:
import pandas as pd
import numpy as np
from pathlib import Path

BASE = Path("../Bases de Dados")

## 1. Carregamento das bases

In [2]:
dfDemitidosTurnoverRH = pd.read_csv(BASE / "FtDemitidosTurnoverRH.csv", sep=";")
dfFuncionarioRH_amostra = pd.read_csv(BASE / "FtFuncionarioRH_amostra.csv", sep=";")

print(f"Demitidos: {dfDemitidosTurnoverRH.shape[0]:,} linhas x {dfDemitidosTurnoverRH.shape[1]} colunas")
print(f"Funcionários (amostra): {dfFuncionarioRH_amostra.shape[0]:,} linhas x {dfFuncionarioRH_amostra.shape[1]} colunas")

Demitidos: 20,587 linhas x 37 colunas
Funcionários (amostra): 24,000 linhas x 27 colunas


## 2. Discovery — Demitidos

In [3]:
print("Colunas:", dfDemitidosTurnoverRH.columns.tolist())
print("\nTipos:")
display(dfDemitidosTurnoverRH.dtypes.to_frame("dtype"))

print("\nNulos (%):")
nulos_d = (dfDemitidosTurnoverRH.isna().mean() * 100).round(1).sort_values(ascending=False)
display(nulos_d[nulos_d > 0].to_frame("pct_nulo"))

Colunas: ['nCodColigada', 'nChapa', 'dAnoMes', 'nIdGestor', 'nAno', 'cMes', 'cCdCCusto', 'cCodSecao', 'cSecao', 'cCargo', 'cFuncao', 'cCdEmpresa', 'cCdEmpresaContabilSeara', 'nCodFilial', 'EMPRESA', 'nCodNivelSal', 'cCodFuncao', 'cEstadoCivil', 'cSexo', 'cFaixasTempoDeCasa', 'cFaixasIdade', 'nIdade', 'cGeracaoNascimento', 'cCor', 'nQtdeDependentes', 'cIniciativaDemissao', 'cDescricaoTipoDemissao', 'cDescricaoMotivoDemissao', 'cEscolaridade', 'cNacionalidade', 'cPosicaoFaixaSalarial', 'cChapa', 'dDataNascimento', 'dDataDemissao', 'dDataAdmissao', 'nSalarioTotal', 'nIdPessoa']

Tipos:


,dtype
nCodColigada,int64
nChapa,int64
dAnoMes,object
nIdGestor,int64
nAno,int64
cMes,object
cCdCCusto,object
cCodSecao,object
cSecao,object
cCargo,object



Nulos (%):


,pct_nulo
cCdEmpresaContabilSeara,32.0
nQtdeDependentes,1.2
cCor,0.2


In [4]:
print("Período (nAno):")
display(dfDemitidosTurnoverRH["nAno"].value_counts().sort_index())

print("\nIniciativa da demissão:")
display(dfDemitidosTurnoverRH["cIniciativaDemissao"].value_counts(dropna=False))

print("\nTop 10 motivos de demissão:")
display(dfDemitidosTurnoverRH["cDescricaoMotivoDemissao"].value_counts().head(10))

Período (nAno):


nAno
2016    2150
2017    1887
2018    1969
2019    2088
2020    2370
2021    3151
2022    3029
2023    3499
2024     444
Name: count, dtype: int64


Iniciativa da demissão:


cIniciativaDemissao
Voluntário      14201
Involuntário     6386
Name: count, dtype: int64


Top 10 motivos de demissão:


cDescricaoMotivoDemissao
Situações Familiares/Pessoais                       5905
Outra proposta de emprego                           3938
Baixa Produtividade                                 2217
Faltas e/ou atrasos                                 1288
Não adaptação as atividades de trabalho              654
Mudança de cidade                                    598
Má conduta                                           533
Faltas e/ou atrasos constantes                       522
Necessidade de cuidados com filhos ou familiares     436
Problemas Familiares/Pessoais                        405
Name: count, dtype: int64

## 3. Discovery — Funcionários (amostra)

In [5]:
print("Colunas:", dfFuncionarioRH_amostra.columns.tolist())
print("\nTipos:")
display(dfFuncionarioRH_amostra.dtypes.to_frame("dtype"))

print("\nNulos (%):")
nulos_f = (dfFuncionarioRH_amostra.isna().mean() * 100).round(1).sort_values(ascending=False)
display(nulos_f[nulos_f > 0].to_frame("pct_nulo"))

Colunas: ['nIdPessoa', 'nCodColigada', 'cFuncao', 'dAnoMes', 'EMPRESA', 'nCodFilial', 'cMes', 'nAno', 'cCargo', 'cSituacao', 'cSecao', 'nNrDependentes', 'cEstadoCivil', 'cSexo', 'cCor', 'cEscolaridade', 'cEstadoEndereco', 'cCidadeEndereco', 'cTurno', 'nTempoDeCasaAnos', 'nIdade', 'cGeracaoNascimento', 'cFaixasTempoDeCasa', 'cFaixasIdade', 'nSalarioTotal', 'nRemuneracaoTotal', 'cPosicaoFaixaSalarial']

Tipos:


,dtype
nIdPessoa,int64
nCodColigada,int64
cFuncao,object
dAnoMes,object
EMPRESA,object
nCodFilial,int64
cMes,object
nAno,int64
cCargo,object
cSituacao,object



Nulos (%):


,pct_nulo
nNrDependentes,90.7
cTurno,32.7
cCor,0.2
cCidadeEndereco,0.1


In [6]:
print("Período (nAno):")
display(dfFuncionarioRH_amostra["nAno"].value_counts().sort_index())

print("\nSituação do colaborador:")
display(dfFuncionarioRH_amostra["cSituacao"].value_counts())

Período (nAno):


nAno
2016    2196
2017    1890
2018    1931
2019    1956
2020    2228
2021    3071
2022    2998
2023    3504
2024    4226
Name: count, dtype: int64


Situação do colaborador:


cSituacao
Demitido                             19639
Ativo                                 3797
Férias                                 300
Af.Previdência                         140
Apos. por Incapacidade Permanente       75
Licença Mater.                          24
Prisão / Cárcere                         7
Licença s/venc                           7
Aviso Prévio                             5
Af.Ac.Trabalho                           4
Doença Ocupacional                       2
Name: count, dtype: int64

## 4. Cruzamento entre bases — definição do universo

Chave de join: `nIdPessoa`

In [7]:
ids_demitidos = set(dfDemitidosTurnoverRH["nIdPessoa"].unique())
ids_funcionarios = set(dfFuncionarioRH_amostra["nIdPessoa"].unique())
intersecao = ids_demitidos & ids_funcionarios
so_demitidos = ids_demitidos - ids_funcionarios
so_funcionarios = ids_funcionarios - ids_demitidos

pessoas_multi_deslig = (dfDemitidosTurnoverRH["nIdPessoa"].value_counts() > 1).sum()

resumo = pd.DataFrame({
    "métrica": [
        "Registros em Demitidos",
        "IDs únicos em Demitidos",
        "Registros em Funcionários (amostra)",
        "IDs únicos em Funcionários (amostra)",
        "IDs em comum",
        "% dos demitidos presentes na amostra",
        "% da amostra presentes em demitidos",
        "IDs só em Demitidos",
        "IDs só em Funcionários",
        "Pessoas com >1 desligamento",
    ],
    "valor": [
        f"{len(dfDemitidosTurnoverRH):,}",
        f"{len(ids_demitidos):,}",
        f"{len(dfFuncionarioRH_amostra):,}",
        f"{len(ids_funcionarios):,}",
        f"{len(intersecao):,}",
        f"{len(intersecao) / len(ids_demitidos):.1%}",
        f"{len(intersecao) / len(ids_funcionarios):.1%}",
        f"{len(so_demitidos):,}",
        f"{len(so_funcionarios):,}",
        f"{pessoas_multi_deslig:,}",
    ],
})

display(resumo)

,métrica,valor
0,Registros em Demitidos,"20,587"
1,IDs únicos em Demitidos,"19,953"
2,Registros em Funcionários (amostra),"24,000"
3,IDs únicos em Funcionários (amostra),"24,000"
4,IDs em comum,"17,939"
5,% dos demitidos presentes na amostra,89.9%
6,% da amostra presentes em demitidos,74.7%
7,IDs só em Demitidos,"2,014"
8,IDs só em Funcionários,"6,061"
9,Pessoas com >1 desligamento,604


## 5. Conclusões preliminares

- **Universo viável:** ~90% dos demitidos estão na amostra de funcionários.
- **Limitação:** ~10% dos demitidos (~2.014 IDs) não aparecem na amostra.
- **Grain pendente:** 604 pessoas têm mais de um registro de desligamento — decidir se analisamos por **pessoa** ou por **evento**.
- **Próximo passo:** usar a interseção (17.939 IDs) como universo inicial e enriquecer com histórico mensal da base de funcionários.